### PR TITLE
add: remote logger request uri sanitisation (JS)

### LINKED ISSUE #50984: [Remote Logging] Replacing the query parameters in URL with some other anonymised string
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Replacing the query parameters in URL with xx or some other anonymized string to prevent log sensitive informations

p1724752577785239/1724752314.673339-slack-C01SFMVEYAK

### PR DESCRIPTION
**Pull request by rjchow**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/50984.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch
3. Go to `Tools -> WCA Test Helper -> Remote Logging`
4. Enable remote logging
5. Click on JS `Simulate Core Exception`, navigate to a wc-admin url with some arbitrary query params such as: `/wp-admin/admin.php?page=wc-admin&key=very_secret_key`
7. Check the log server page for the log entry (See test plan in D155228-code to see how to check the log entry)
8. Confirm the log properties include `request_uri` and the non-whitelisted parameters have been masked
### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by chihsuan — COMMENTED** [review]
Nice work! Tested well. Thanks for updating the doc. 👍

**Review by chihsuan — APPROVED** [review]

File: packages/js/remote-logging/README.md [hunk 1]
```diff
@@ -172,3 +172,27 @@ addFilter(
...
+
+**Usage example:**
+
+```js
+import { addFilter } from '@wordpress/hooks';
+
+addFilter(
+  'woocommerce_remote_logging_request_uri_whitelist',
+  'my-plugin',
+  (whitelist) => {
```
Comment (psealock) [inline packages/js/remote-logging/README.md]: Sorry... so nit picky

```suggestion
  ( whitelist ) => {
```

File: packages/js/remote-logging/src/remote-logger.ts [hunk 2]
```diff
@@ -33,6 +33,39 @@ export const REMOTE_LOGGING_LOG_ENDPOINT_FILTER =
...
+export const REMOTE_LOGGING_REQUEST_URI_PARAMS_WHITELIST_FILTER =
+	'woocommerce_remote_logging_request_uri_whitelist';
+
+export const REMPOTE_LOGGING_REQUEST_URI_PARAMS_DEFAULT_WHITELIST = [
+	'path',
+	'page',
+	'step',
+	'task',
+	'tab',
+];
```
Comment (psealock) [inline packages/js/remote-logging/src/remote-logger.ts:50]: * `section` - ie, "/wp-admin/admin.php?page=wc-settings&tab=shipping&section=classes"
* `status` - ie, "/wp-admin/admin.php?page=wc-status&tab=action-scheduler&status=complete"
* `post_type` - ie, "/wp-admin/edit.php?post_type=product"
* `taxonomy` - ie, "/wp-admin/edit-tags.php?taxonomy=product_cat&post_type=product"
* `action` - ie, "/wp-admin/admin.php?page=wc-orders&action=new" or "/wp-admin/admin.php?page=wc-orders&action=edit&id=61"

**Review by psealock — COMMENTED** [review]
This is testing well. Just a nit pick and some additional query params we could add to the whitelist. These are all for PHP pages, so I guess they aren't relevant?

File: packages/js/remote-logging/src/remote-logger.ts (hunk 2 above)
Comment (psealock) [reply packages/js/remote-logging/src/remote-logger.ts:50]: Does the PHP side of things also send query parameters?

Comment (rjchow) [reply packages/js/remote-logging/src/remote-logger.ts:50]: There's a PHP side of the implementation here: https://github.com/woocommerce/woocommerce/pull/51013
I think I'd like the whitelist to be homogenous so I'll amend both sides to include the same query params. Thanks for the suggestions!

**Review by psealock — APPROVED** [review]
Thanks for the changes, looks great 🚢

**Pull request merged**